### PR TITLE
Add GH action to automatically run Coverity scans

### PR DESCRIPTION
**Pull request by jamessan**
This requires adding two new secrets to the [repository settings](https://github.com/vim/vim/settings/secrets/actions).

* COVERITY_SCAN_TOKEN
* COVERITY_SCAN_EMAIL

These are the token and email from the curl command on https://scan.coverity.com/projects/vim/builds/new.

### CODE REVIEW

File: .github/workflows/coverity.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          sudo update-alternatives --install /usr/bin/lua lua /usr/bin/lua5.3 10
+
+      - name: Configure
+        run: |
+          ./configure --with-features=huge ${CONFOPT} --enable-fail-if-missing
+          # Append various warning flags to CFLAGS.
+          sed -i -f ci/config.mk.sed ${SRCDIR}/auto/config.mk
+          sed -i -f ci/config.mk.${CC}.sed ${SRCDIR}/auto/config.mk
+
+      - name: Build/scan neovim
```
Comment (chrisbra) [inline .github/workflows/coverity.yml]: > Build/scan neovim

Vim

**Comment by chrisbra** [comment]
For what its worth, I have setup a daily coverity cronjob job, that submits a build. I have not been run into the submission limit.

**Comment by chrisbra** [comment]
I setup those two new tokens

**Review by chrisbra — APPROVED** [review]

**Comment by jamessan** [comment]
> For what its worth, I have setup a daily coverity cronjob job, that submits a build. I have not been run into the submission limit.

Yeah, at Vim's current size, 2 builds/day are allowed.  I was just giving some buffer.  I've changed it to every day, since that should still avoid the limit for quite a long time.

**Comment by brammool** [comment]
It fails to run:
```
config.status: creating auto/config.mk
config.status: creating auto/config.h
sed: can't read /auto/config.mk: No such file or directory
Error: Process completed with exit code 2.
```

Apparently $SRCDIR is not set:
```
 sed -i -f ci/config.mk.sed ${SRCDIR}/auto/config.mk
```

**Pull request closed without merging**